pkg/controller/jobs/statefulset/statefulset_reconciler.go

```diff
@@ -170,8 +182,18 @@ func (r *Reconciler) handle(obj client.Object) bool {
 	if !isSts {
 		return true
 	}
+
+	ctx := context.Background()
+	log := r.log.WithValues("statefulset", klog.KObj(sts))
+	ctrl.LoggerInto(ctx, log)
+
 	// Handle only statefulset managed by kueue.
-	return jobframework.IsManagedByKueue(sts)
+	suspend, err := jobframework.WorkloadShouldBeSuspended(ctx, sts, r.client, r.manageJobsWithoutQueueName, r.managedJobsNamespaceSelector)
+	if err != nil {
+		log.Error(err, "Failed to determine suspend state")
```

Comment:
```suggestion
		log.Error(err, "Failed to determine if the StatefulSet should be managed by Kueue")
```
We are not determining the suspend state. We are determining if we should manage - if we manage then we change the suspend state accordingly.